public/app/plugins/datasource/prometheus/dataquery.cue

```diff
@@ -0,0 +1,54 @@
+// Copyright 2023 Grafana Labs
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package grafanaplugin
+
+import (
+	common "github.com/grafana/grafana/packages/grafana-schema/src/common"
+	"github.com/grafana/grafana/pkg/plugins/pfs"
+)
+
+// This file (with its sibling .cue files) implements pfs.GrafanaPlugin
+pfs.GrafanaPlugin
+
+composableKinds: DataQuery: {
+	maturity: "merged"
+
+	lineage: {
+		seqs: [
+			{
+				schemas: [
+					{
+						common.DataQuery
+						expr:            string
+						format?:         string
```

Comment:
What are the valid values?  I think 'timeseries' and 'table', but would be nice to have that explicit :)

Reply:
Agree, if there's a set of acceptable values here, they should be explicitly declared. 

Could make it a reference to another type, as with `#QueryEditorMode`, if that type needs to be used elsewhere in TS

Reply:
I will check the possible values and introduce a change.

Reply:
See https://github.com/grafana/grafana/pull/63878/commits/a0bd6216c63b03672a39ed102f498cc213514b49